openapi/code/service.go

```diff
@@ -131,20 +131,66 @@ var crudNames = map[string]bool{
 	"restore": true,
 }
 
-func (svc *Service) newRequest(params []openapi.Parameter, op *openapi.Operation) *Entity {
-	if op.RequestBody == nil && len(params) == 0 {
-		return nil
+func (svc *Service) updateEntityTypeFromMimeType(entity *Entity, mimeType openapi.MimeType) {
+	if mimeType == "" || mimeType == openapi.MimeTypeJson {
+		return
+	}
+	// For request or response bodies that are not application/json, the body
+	// is modeled by a byte stream.
+	entity.IsByteStream = true
+	entity.IsEmpty = false
+	entity.IsAny = false
+}
+
+// Construct the base request entity for a given operation. For requests whose
+// mime type is not application/json, the request body is always a byte stream.
+// For requests whose mime type is application/json, the request body consists
+// of the top-level fields of the request object as defined in the OpenAPI spec.
+// Additionally, if the x-databricks-body-field-name extension is specified,
+// the request body is nested into a field with that name. This is necessary for
+// requests that are not application/json but that have additional request
+// parameters.
+func (svc *Service) newMethodEntity(op *openapi.Operation) (*Entity, openapi.MimeType, *Field) {
+	if op.RequestBody == nil {
+		return &Entity{fields: map[string]*Field{}}, "", nil
 	}
-	request := &Entity{fields: map[string]*Field{}}
-	if op.RequestBody != nil {
-		request = svc.Package.schemaToEntity(op.RequestBody.Schema(), []string{op.Name()}, true)
+	mimeType, mediaType := op.RequestBody.MimeTypeAndMediaType()
+	requestSchema := mediaType.GetSchema()
+	if requestSchema == nil {
+		return &Entity{fields: map[string]*Field{}}, "", nil
 	}
-	if request == nil {
+	// If x-databricks-body-field-name is specified, the request structure
+	// contains a field with that name whose value is the request body.
+	if mimeType.IsByteStream() {
+		requestSchema = &openapi.Schema{
```

Comment:
let's use this logic in responses as well rather than returning bare bytestreams.